packages/mui-material/src/Chip/Chip.js

```diff
@@ -103,6 +103,20 @@ const ChipRoot = styled('div', {
       padding: 0, // Remove `button` padding
       verticalAlign: 'middle',
       boxSizing: 'border-box',
+
+      // gggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggg
+      '& > *': {
+        cursor: 'inherit', // Inherit cursor from parent
+      },
+      '&.clickable': {
+        cursor: 'pointer',
+        userSelect: 'none',
+        WebkitTapHighlightColor: 'transparent',
+      },
+      '&.clickable > *': {
+        pointerEvents: 'none', // Prevent nested elements from capturing pointer events
+      },
+      // gggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggg
```

Comment:
I don't think all this logic is needed.  We can simply reset inherited `line-height` style using `line-height: normal`:
```diff
--- a/packages/mui-material/src/Chip/Chip.js
+++ b/packages/mui-material/src/Chip/Chip.js
@@ -89,6 +89,7 @@ const ChipRoot = styled('div', {
       alignItems: 'center',
       justifyContent: 'center',
       height: 32,
+      lineHeight: 'normal',
       color: (theme.vars || theme).palette.text.primary,
       backgroundColor: (theme.vars || theme).palette.action.selected,
       borderRadius: 32 / 2,
```
Docs: https://developer.mozilla.org/en-US/docs/Web/CSS/line-height#normal